pyvera/__init__.py

```diff
@@ -731,12 +716,22 @@ def is_locked(self, refresh=False):
 class VeraThermostat(VeraDevice):
     """Class to represent a thermostat."""
 
-    def get_payload_parameter_name(self, name):
-        return "New" + name
-
     def set_temperature(self, temp):
         """Set current goal temperature / setpoint"""
-        self.set_value('CurrentSetpoint', temp)
+
+        # Set both the heat and cool setpointa
+        self.set_service_value(
+            self.thermostat_cool_setpoint,
+            'CurrentSetpoint',
+            'NewCurrentSetpoint',
+            mode)
+
+        self.set_service_value(
+            self.thermostat_heat_setpoint,
+            'CurrentSetpoint',
+            'NewCurrentSetpoint',
+            mode)
```

Comment:
I'm not sure this will always be the correct action to take. I've been working through some confusion with the Thermostat API with Vera support, and I'm still a bit confused TBH. Here are some quotes from our support thread:

> Auto is not a mode which you should set up manually, would be better if you could set your Heat and Cool setpoints and then leave Auto to work on it’s own making the average and running on that value.

> Also, in order to work properly, for some thermostats the different between Heat and Cool must be greater than 3 degrees, please try to enlarge the gap between Heat and Cool and then give Auto a try.

In a non-auto mode, setting `CurrentSetpoint` seems like the appropriate action to take. When a thermostat is in Auto mode, it almost sounds like they want the following actions to take place to make a change:

* Set the operating mode to some non-auto value
* Update the heat and cool values, making sure they're least 3 degrees apart
* Set the operating mode back to the existing mode

So, in short, this isn't right, but it's clear to me that I need to do some more testing before I'm able to tell you what to change it to.